tket/src/Circuit/include/Circuit/Circuit.hpp

```diff
@@ -1726,6 +1741,8 @@ Vertex Circuit::add_op(
     }
   }
 
+  add_wasm_register(count_wasm_sig);
```

Comment:
If `count_wasm_sig` is 0 this will do nothing and that is the behaviour I had in mind. I think it would be good not to have this on circuits where we have no wasm operations?